src/main/java/gdsc/shine/springlearningsimple/controllers/BeanController.java

```diff
@@ -0,0 +1,22 @@
+package gdsc.shine.springlearningsimple.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.Arrays;
+import java.util.List;
+
+@RestController
+public class BeanController {
+    private final ApplicationContext ac;
+
+    public BeanController(ApplicationContext ac) {
+        this.ac = ac;
+    }
+    @GetMapping("/beans")
+    public List<String> getBeans() {
+        return Arrays.asList(ac.getBeanDefinitionNames());
```

Comment:
Arrays.asList와 List.of는 뭐가 다를까요?

Reply:
`Arrays.asList()`를 사용하면 변환된 리스트의 길이가 고정되는 것으로 알고있었습니다. 컨트롤러가 Bean 값들을 수정하지 않을 것 같아 추가/삭제가 불가능한 리스트로 리턴하는 방향으로 작성했습니다.

그런데 찾아보니 `List.of()` 메소드는 리스트의 길이가 고정되고, null값을 받지 않으며 수정도 불가하네요!
Arrays.asList()는 리스트의 사이즈 변동은 안되지만 변경은 가능하고 Null값도 받을 수 있습니다.

- 좀더 극명한  차이는 두 메소드가 반환하는 리스트가 직렬화 여부로 갈리기도 하네요.
```
Container class for immutable collections. Not part of the public API. Mainly for namespace management and shared infrastructure. Serial warnings are suppressed throughout because all implementation classes use a serial proxy and thus have no need to declare serialVersionUID.
```
List.of 같은 경우에는 직렬화가 안되는 ` immuable collection`이기 때문인데... 
어떻게 서버기능이 추가되고 설계가 달라지느냐에 따라 자바 객체 자체를 직렬화할 수도 안할 수도 있지만 가능성을 열어두는 게 좋을 것 같습니다.

Reply:
추가, 삭제를 아예 막고싶다면 Collections.unmodifiableList(list); 사용 추천해봅니다~